### PR TITLE
add cops to chunk management tests

### PR DESCRIPTION
**Pull request by matklad**
Follow up to #7280 

closes https://nearinc.atlassian.net/browse/CPR-141

### CODE REVIEW

File: integration-tests/src/tests/client/chunks_management.rs [hunk 1]
```diff
@@ -297,7 +326,74 @@ fn chunks_recovered_from_full() {
...
+        run_actix(async {
+            chunks_produced_and_distributed_common(
+                4,
+                true,
+                &["test1", "cop1"],
+                4 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
+            );
+        });
+    });
+}
```
Comment (matklad) [inline integration-tests/src/tests/client/chunks_management.rs:385]: @mzhangmzz so this one baffles me: `chunks_recovered_from_full_timeout_too_short` (without cops) is a `should_panic` test we expect it to fail.

But somehow, if I add cops, it actually passes. 

I don't have a great understanding of what *actually* should be happening here.

**Comment by matklad** [comment]
While looking closer at the tests, I've found one potential explanation I think!

In this line

https://github.com/near/nearcore/blob/525a87764dd8081228a1aeb24d611ae058474490/chain/chunks/src/lib.rs#L596

in `request_partial_encoded_chunk` we are picking some random validator to request a chunk from. *But* we are considering only block producers, not chunk-only producers. Is this expected?

**Comment by matklad** [comment]
rebased

**Comment by mzhangmzz** [comment]
> While looking closer at the tests, I've found one potential explanation I think!
> 
> In this line
> 
> https://github.com/near/nearcore/blob/525a87764dd8081228a1aeb24d611ae058474490/chain/chunks/src/lib.rs#L596
> 
> in `request_partial_encoded_chunk` we are picking some random validator to request a chunk from. _But_ we are considering only block producers, not chunk-only producers. Is this expected?

Yeah we should also consider chunk only producer there, but I think that won't cause the issue here because the problem with this test is that the chunk shouldn't be distributed successfully. I took a look at the code and couldn't see anything obviously wrong. I'll need to run the test and see what's going on from the log, which I'll do on Monday.

**Comment by matklad** [comment]
Commented out the `should_panic` for now, so that this is landable (I think it's an OK process to commit a test with a *wrong* outcome, if it has a clear fixme. But we can also debug this here).

**Comment by matklad** [comment]
Filed https://nearinc.atlassian.net/browse/CPR-144 for `get_random_target_tracking_shard`

File: integration-tests/src/tests/client/chunks_management.rs [hunk 2]
```diff
@@ -298,11 +321,80 @@ fn chunks_recovered_from_full_timeout_too_short() {
...
+// #[should_panic]
+#[cfg(feature = "protocol_feature_chunk_only_producers")]
+fn chunks_recovered_from_full_timeout_too_short_cop() {
     heavy_test(|| {
         run_actix(async {
             chunks_produced_and_distributed_common(
                 4,
                 true,
+                &["test1", "cop1"],
+                4 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
```
Comment (mzhangmzz) [inline integration-tests/src/tests/client/chunks_management.rs:382]: @matklad This line should be `2 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS`. If the block delay is too large, then test4 have enough time to reconstruct full chunk from the parts. The test does panic when I changed the value.

Comment (matklad) [reply integration-tests/src/tests/client/chunks_management.rs]: Thanks for looking into this, updated, ptal!

**Review by mzhangmzz — APPROVED** [review]

**Comment by mzhangmzz** [comment]
Thank you!

**Pull request merged**